doc/source/development/rfc/rfc92_wkb_only_geometries.rst

```diff
@@ -0,0 +1,228 @@
+that use WKB natively.
+
+Technical details
+-----------------
+
+The proposal mostly contains in adding a new subclass of OGRGeometry called
+`OGRWKBOnlyGeometry`
+
+.. code-block:: c++
+
+    /**
+     * Special OGRGeometry subclass that only holds its WKB representation.
+     *
+     * Used for optimizations when passing geometries between drivers or
+     * application code that does not require to query details of the geometry,
+     * such as structure in sub-components or vertex coordinates.
+     */
+    class CPL_DLL OGRWKBOnlyGeometry final: public OGRGeometry
+    {
+        std::vector<GByte> m_abyWKB{};
+        OGREnvelope        m_sEnvelope{}; // optional
+
+    public:
+        OGRWKBOnlyGeometry(const void* pabyWKB, size_t nWKBSize);
+        OGRWKBOnlyGeometry(const void* pabyWKB, size_t nWKBSize, const OGREnvelope& sEnvelope);
+
+        /** Get WKB */
+        const std::vector<GByte>& Wkb() const { return m_abyWKB; }
+
+        /** Return a "real" OGRGeometry instantiated from the WKB */
+        std::unique_ptr<OGRGeometry> Materialize() const;
+
+        /** Returns the equivalent of Materialize()->getGeometryType() without
+         * materializing */
+        OGRwkbGeometryType getUnderlyingGeometryType() const override;
+
+        /** Returns envelope stored at construction time, or "quickly"
+         * determined by inspecting the WKB content */
+        void getEnvelope(OGREnvelope *psEnvelope) const override;
+
+        // Dummy implementation of all pure virtual methods of OGRGeometry
+        // ==> all return an error
+        // Typically getGeometryType() returns wkbUnknown, to avoid user code
+        // to wrongly cast to a OGRPoint/OGRLineString/etc. instance.
+    };
+
+
+A new method is added to the OGRLayer class:
+
+.. code-block:: c++
+
+    /** If bRequestWKBOnlyGeometries is true, then the driver should return, in
+     *  GetNextFeature(), geometries that are instance of OGRWKBOnlyGeometry.
+     *  Only drivers for which TestCapability(OLCReadWKBGeometries) is true
+     *  are capable of this. Other drivers will error out.
+     */
+    virtual OGRErr RequestWKBOnlyGeometries(bool bRequestWKBOnlyGeometries);
+
+Two new capabilities are added at the OGRLayer level:
+
+* OLCReadWKBGeometries: a layer must return TRUE for it when the layer can
+  honour RequestWKBOnlyGeometries(true)
+* OLCWriteWKBGeometries: a layer must return TRUE for it if its CreateFeature()
+  and SetFeature() implementations support being passed OGRWKBOnlyGeometry
+  instances.
+
+ogr2ogr is modified to call RequestWKBOnlyGeometries(true) on the source layer:
+
+* if the source layer advertises OLCReadWKBGeometries
+* if the target layer advertises OLCWriteWKBGeometries
+* if no command line switch requires a "materialized" geometry.
+
+So basically this is for requests like
+"ogr2ogr out.gpkg in.gpkg [layer or SQL request] [attribute filter] [spatial filter]"
```

Comment:
Wouldn't spatial filters often require full geometry parsing (assuming the filter can't be passed to a backend/read from an index only)

Reply:
note added